cmd/catalyst-node/catalyst-node.go

```diff
@@ -478,11 +497,84 @@ func redirectHandler(redirectPrefixes []string, nodeHost string) http.Handler {
 
 		rPath := fmt.Sprintf(pathTmpl, fullPlaybackID)
 		rURL := fmt.Sprintf("%s://%s%s", protocol(r), bestNode, rPath)
+		rURL, err = resolveNodeURL(rURL)
+		if err != nil {
+			glog.Errorf("failed to resolve node URL playbackID=%s err=%s", playbackID, err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 		glog.V(6).Infof("generated redirect url=%s", rURL)
 		http.Redirect(w, r, rURL, http.StatusFound)
 	})
 }
 
+func streamSourceHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		b, err := io.ReadAll(r.Body)
+		if err != nil {
+			glog.Errorf("error handling STREAM_SOURCE body=%s", err)
+			w.Write([]byte("push://"))
+			return
+		}
+		streamName := string(b)
+		glog.V(7).Infof("got mist STREAM_SOURCE request=%s", streamName)
+		dtscURL, err := queryMistForClosestNodeSource(streamName, "0", "0", "", true)
```

Comment:
I'm not sure I understand why you query here Mist with lat/lon as `0/0`, could you maybe add some comment with the explanation?